src/components/budget-expiry-notification/data/hooks/useExpirationMetadata.jsx

```diff
@@ -0,0 +1,40 @@
+import dayjs from 'dayjs';
+import { useMemo } from 'react';
+import useExpiryThresholds from './useExpiryThresholds';
+import { PLAN_EXPIRY_VARIANTS } from '../constants';
+
+const useExpirationMetadata = (endDateStr = 'January 1, 1970') => {
```

Comment:
[concern/suggestion] I'm a bit concerned about the default fallback argument to January 1, 1970 when `endDateStr` is not provided. If a customer has no Learner Credit (e.g., subscriptions only customers), this hook will consider it as having a Learner Credit budget available with an end date, which is not technically correct.

The only conditional logic preventing expiration messaging (alert+modal) to appear is the `durationDiff.asDays() >= 0` check below. Without it, we risk showing Learner Credit expiration messaging without the customer having any Learner Credit.

I might recommend a slight refactor to not solely on `durationDiff.asDays() >= 0`, but allow `endDateStr` to be `undefined` in the below logic, e.g.:

```jsx
return useMemo(() => {
  const emptyExpirationMetadata = {
     thresholdKey: null,
      threshold: null,
      isPlanApproachingExpiry: false,
  };
  if (!endDateStr) {
    return emptyExpirationMetadata;
  }
  const endDate = dayjs(endDateStr);
  const today = dayjs();
  const durationDiff = dayjs.duration(endDate.diff(today));
  const thresholdKeys = Object.keys(expiryThresholds).sort((a, b) => a - b);
  const thresholdKey = thresholdKeys.find((key) => durationDiff.asDays() >= 0 && durationDiff.asDays() <= key);
  if (!thresholdKey) {
    return emptyExpirationMetadata;
  }

  // ...
}, [endDateStr, expiryThresholds]);
```